test_conformance/non_uniform_work_group/TestNonUniformWorkGroup.cpp

```diff
@@ -706,12 +706,38 @@ size_t TestNonUniformWorkGroup::adjustGlobalBufferSize(size_t globalBufferSize)
   size_t adjustedGlobalBufferSize = globalBufferSize;
   if (deviceMaxAllocObjSize < globalBufferSize) {
     adjustedGlobalBufferSize = deviceMaxAllocObjSize;
-    log_info("globalBufferSize was adjusted from %lu to %lu\n", globalBufferSize, adjustedGlobalBufferSize);
   }
 
   return adjustedGlobalBufferSize;
 }
 
+void TestNonUniformWorkGroup::adjustGlobalSize(size_t maxNumOfGlobalWorkItems)
+{
+    size_t oldNumOfGlobalWorkItems = _numOfGlobalWorkItems;
+    size_t oldGlobalSize[MAX_DIMS];
+    memcpy(oldGlobalSize, _globalSize, sizeof(_globalSize));
+    // Adjusting _globalSize[1] and _globalSize[2] is enough.
+    // _globalSize[0] is unchanged because it controls non-uniform workgroup.
+    for (int i = (int)_dims - 1; i >= 1; --i)
+    {
+        while (_numOfGlobalWorkItems > maxNumOfGlobalWorkItems
+               && _globalSize[i] >= (2 * _localSize[i]))
+        {
+            _globalSize[i] /= 2;
+            _numOfGlobalWorkItems =
+                _globalSize[0] * _globalSize[1] * _globalSize[2];
+        }
+    }
+    assert(oldNumOfGlobalWorkItems != _numOfGlobalWorkItems
```

Comment:
Not sure if such assert is acceptable. Maybe better solution is to return error code and modify variables by reference. Notice that `adjustGlobalBufferSize` function can return error code as `globalBufferSize`. It could be fixed as well when you decide to make one common function

Reply:
assert is removed